### PR TITLE
new deep river dock

### PR DESCRIPTION
**Pull request by bobchaos**
#### Summary
SUMMARY: Content "Add a deep river dock and associated construction recipe"

#### Purpose of change
This change introduces a new, very long to craft deep river dock, intended for use in river base expansions and probably mapgen (Think local marina and such)

#### Describe the solution
N/A

#### Describe alternatives you've considered
Better values? I found inspiration in other recipes but I know next to nothing about construction, this could be profoundly unrealistic for all I know. If anyone has better components and times to suggest I'm open to it. The pile I'm especially worried about, as I'm not quite sure how those were setup prior to heavy machinery and modern caissoning techniques.

I've also considered adding an alternative concrete pile as a first step, which would be much faster but require things like a mechanical pump and 2x4 to create a caisson, drain it and pour concrete, but I'm not sure it can be done without adding a new type of tool quality (pumping, presumably).

#### Additional context
I plan on using this in the very near future for a fishing dock basecamp expansion.

### CODE REVIEW

**Comment by curstwist** [comment]
Please add `looks_like` entries for your new terrains.

**Comment by bobchaos** [comment]
> Please add `looks_like` entries for your new terrains.

I could make the final product look like a pontoon bridge, but what about the frame and pile? Not sure there's anything fitting :/ ideas?

**Comment by bobchaos** [comment]
I added some "looks_like", but I feel the pile may look a bit silly and unintuitive: It's a deep water tile that looks like a column :O

Are we sure we don't prefer to leave the pile with no graphics? It would default to "|", which at least looks somewhat appropriate.

File: data/json/furniture_and_terrain/terrain-bridges-docks.json [hunk 1]
```diff
@@ -147,4 +211,4 @@
       "items": [ { "item": "rock", "count": [ 4, 20 ] }, { "item": "rebar", "count": [ 10, 20 ] } ]
     }
   }
-]
+]
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-bridges-docks.json]:
```suggestion
]

```

File: data/json/construction.json [hunk 2]
```diff
@@ -3089,4 +3125,4 @@
     "pre_flags": [ "FLAT" ],
     "post_terrain": "t_radio_controls"
   }
-]
+]
```
Comment (ZhilkinSerg) [inline data/json/construction.json]:
```suggestion
]

```

**Comment by bobchaos** [comment]
I commited windows line endings didn't I? I _really_ need to take some time a setup a proper workstation. Sorry :/

**Comment by ZhilkinSerg** [comment]
> 
> 
> I commited windows style didn't I? I _really_ need to take some time a setup a proper workstation. Sorry :/

You can use [web-linter](http://dev.narc.ro/cataclysm/format.html) to format your json files properly.

**Comment by bobchaos** [comment]
> 
> 
> > I commited windows style didn't I? I _really_ need to take some time a setup a proper workstation. Sorry :/
> 
> You can use [web-linter](http://dev.narc.ro/cataclysm/format.html) to format your json files properly.

I did, and then pasted right back into notepad++ and used git for windows that messes with the line endings and final line feed.... I'll review my settings for the next one

**Pull request merged**